hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -698,28 +671,20 @@ contestTx scriptRegistry vk Snapshot{number, utxo, utxoToDecommit, version} sig
       Head.Closed
         Head.ClosedDatum
           { snapshotNumber = toInteger number
-          , utxoHash
-          , deltaUTxOHash
+          , utxoHash = toBuiltin $ hashUTxO @Tx utxo
+          , deltaUTxOHash =
+              case contestRedeemer of
+                Head.ContestCurrent{} ->
+                  Just . toBuiltin $ hashUTxO @Tx $ fromMaybe mempty utxoToDecommit
+                _ -> Nothing
           , parties = closedParties
           , contestationDeadline = newContestationDeadline
           , contestationPeriod = onChainConstestationPeriod
           , headId = headIdToCurrencySymbol headId
           , contesters = contester : closedContesters
-          , version = toInteger version
+          , version = toInteger openVersion -- TODO: This was 'version' before and that was wrong. Which test can catch it?
```

Comment:
This would be an interesting test to write still.

Reply:
Resolved in https://github.com/cardano-scaling/hydra/pull/1513/commits/df2fcf2b7d0f5c08ec15dbfd4a2cd3a0711b76f0